### PR TITLE
Move fetch-latest-release to be local to content-build.

### PR DESCRIPTION
**Pull request by timcosgrove**
## Description
relates to https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12640

This is a port of [fetch-latest-release](https://github.com/gregziegan/fetch-latest-release) into the content-build repo. This has been done to update dependencies and syntax, as the original action appears to be abandoned by its maintainer.

As near as I can tell, this in the only place is the VA ecosystem this action is used, and so making this a local action seems the best course of action.

## Testing done & Screenshots
Test run using the action: https://github.com/department-of-veterans-affairs/content-build/actions/runs/4824297324/jobs/8593897277#step:5:18

Confirmed that it retrieved the most recent valid release: https://github.com/department-of-veterans-affairs/content-build/commits/v0.0.3248

## Acceptance criteria

- [ ] Action performs its function without failure

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: .github/workflows/authenticated-cms-request/action.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ inputs:
   timeout:
     description: Request timeout
     required: false
-    default: 10000
+    default: "10000"
```
Comment (timcosgrove) [inline .github/workflows/authenticated-cms-request/action.yml:24]: This is unrelated to the rest of the PR; just, GHA technical wants a string here.

**Review by ndouglas — APPROVED** [review]

**Comment by timcosgrove** [comment]
These changes were rolled into https://github.com/department-of-veterans-affairs/content-build/pull/1558
